buildpack/telemetry/appdynamics.py

```diff
@@ -18,6 +18,63 @@
     "machine-agent",
 )
 
+APPDYNAMICS_NODE_NUMBER = os.getenv("CF_INSTANCE_INDEX", default=0)
+APPDYNAMICS_TIER = util.get_vcap_data()["application_name"]
+
+# Number suffix will be added to AppDynamics Node Name
+# The key will be used in the function '_filter_default_env'
+APPDYNAMICS_NODE_NAME_KEY = "APPDYNAMICS_AGENT_NODE_NAME"
+
+APPDYNAMICS_DEFAULT_ENV_VARS = {
+    "APPDYNAMICS_AGENT_APPLICATION_NAME": util.get_app_from_domain(),
+    APPDYNAMICS_NODE_NAME_KEY: "node",  # This is 'base' name without suffix
+    "APPDYNAMICS_AGENT_TIER_NAME": APPDYNAMICS_TIER,
+    "APPDYNAMICS_CONTROLLER_PORT": "443",
+    "APPDYNAMICS_CONTROLLER_SSL_ENABLED": "true",
+}
+
+
+def _filter_default_env():
+    """
+    The function compares environment variables which are set up by user and
+    the initial list of default variables (APPDYNAMICS_DEFAULT_ENV_VARS).
+
+    A variable which is set up by user has priority in case of matching by name.
+
+    """
+
+    os_env_dict = dict(os.environ)
+
+    filtered_default_env_vars = {}
+
+    node_name = os_env_dict.get(APPDYNAMICS_NODE_NAME_KEY)
+
+    if node_name is None:
+        node_name = APPDYNAMICS_DEFAULT_ENV_VARS.get(APPDYNAMICS_NODE_NAME_KEY)
+
+    #  Add suffix to AppDynamics Node Name
+    filtered_default_env_vars[
+        APPDYNAMICS_NODE_NAME_KEY
+    ] = f"{node_name}_{APPDYNAMICS_NODE_NUMBER}"
+
+    for var_name, value in APPDYNAMICS_DEFAULT_ENV_VARS.items():
+
+        if var_name == APPDYNAMICS_NODE_NAME_KEY:
+            continue
+
+        if not (var_name in os_env_dict):
+            filtered_default_env_vars[var_name] = value
+
+    return filtered_default_env_vars
+
+
+FILTERED_APPDYNAMICS_DEFAULT_ENV_VARS = _filter_default_env()
```

Comment:
Why do we need to assign it to a variable and from the [all caps style](https://peps.python.org/pep-0008/#constants) it suggests it's a constant, which it is not.

Reply:
Doesn't exist anymore